crates/biome_grit_parser/src/lexer/mod.rs

```diff
@@ -180,7 +180,7 @@ impl<'src> GritLexer<'src> {
     fn lex_token(&mut self, current: u8) -> GritSyntaxKind {
         match current {
             b'\n' | b'\r' => {
-                debug_assert!(self.consume_newline());
+                assert!(self.consume_newline());
```

Comment:
Actually, I think we can remove the `assert!` call altogether. We have test assertions that check if the lexer is not progressing.

```suggestion
                self.consume_newline();
```